community/data-structure-on-rocksdb.md

```diff
@@ -314,3 +315,28 @@ where the `payload` is a string encoded in the corresponding `format`:
 | CBOR       |          1 |
 
 Also, if we decide to add a more IO-friendly format to avoid reading all payload to the memory before searching an element via JSONPath or seperate a relatively large JSON to multiple key-values, we can take advantage of the `format` field.
+
+## Hyperloglog
+
+Redis hyperloglog can be thought of as a static array with a length of 16384. The array elements are called registers, which are used to store the maximum count of consecutive 0s. This register array is the input parameter for the hyperloglog algorithm. 
+In Kvrocks, the hyperloglog data structure is stored in following two parts:
+ 
+#### hyperloglog metadata
+
+```text
+        +----------+------------+-----------+-----------+
+key =>  |  flags   |  expire    |  version  |  size     |
+        | (1byte)  | (Ebyte)    |  (8byte)  | (Sbyte)   |
+        +----------+------------+-----------+-----------+
+```
+- `size` records the number of registers, which is a constant, even no element is added.
+
+#### hyperloglog sub keys-values
+
+```text
+                              +---------------+
+key|version|register_index => |     0s count  |
```

Comment:
we can denote the format of count. In current implementation, the count is an `count-to-string` ?